csharp/src/Ice/LocalException.cs

```diff
@@ -2,183 +2,261 @@
+    /// <summary>This exception reports an attempt to use a destroyed communicator.</summary>
+    [Serializable]
+    public class CommunicatorDestroyedException : ObjectDisposedException
     {
-        public string Name;
-
-        public ObjectAdapterDeactivatedException() => Name = "";
-
-        public ObjectAdapterDeactivatedException(System.Exception ex) : base("", ex) => Name = "";
-
-        public ObjectAdapterDeactivatedException(string name) => Name = name;
+        public CommunicatorDestroyedException() :
+            base("")
+        {
+        }
 
-        public ObjectAdapterDeactivatedException(string name, System.Exception ex) : base("", ex) => Name = name;
+        public CommunicatorDestroyedException(Exception ex)
+            : base("", ex)
+        {
+        }
 
+        protected CommunicatorDestroyedException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+        }
     }
 
-    /// <summary>
-    /// This exception is raised if no suitable endpoint is available.
-    /// </summary>
-    public class NoEndpointException : System.Exception
+    /// <summary>This exception reports an attempt to use a deactivated object adapter.</summary>
+    [Serializable]
+    public class ObjectAdapterDeactivatedException : ObjectDisposedException
     {
-        public string Proxy;
-
-        public NoEndpointException() => Proxy = "";
+        public ObjectAdapterDeactivatedException(string objectName)
+            : base(objectName)
+        {
+        }
 
-        public NoEndpointException(System.Exception ex) : base("", ex) => Proxy = "";
+        protected ObjectAdapterDeactivatedException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+        }
+    }
 
-        public NoEndpointException(string proxy) => Proxy = proxy;
+    /// <summary>This exception reports that a proxy's endpoints could not be resolved./// </summary>
+    [Serializable]
+    public class NoEndpointException : Exception
+    {
+        public NoEndpointException() // TODO: remove this constructor
+        {
+        }
 
-        public NoEndpointException(string proxy, System.Exception ex) : base("", ex) => Proxy = proxy;
+        public NoEndpointException(string stringifiedProxy)
+            : base($"could not find the endpoints for proxy `{stringifiedProxy}'")
+        {
+        }
 
+        protected NoEndpointException(SerializationInfo info, StreamingContext context)
+            : base(info, context)
+        {
+        }
     }
 
-    /// <summary>
-    /// This exception indicates socket errors.
-    /// </summary>
-    public class SocketException : System.Exception
+    /// <summary>This exception reports an error from the transport layer.</summary>
+    [Serializable]
+    public class TransportException : Exception
     {
-        public SocketException()
+        public TransportException()
```

Comment:
Can we remove this constructor, it will be good that we always provide a message and/or inner exception

Reply:
I made it protected and fixed the corresponding code. Removing it entirely it not doable since we throw many of the derived exceptions without additional info.